### DISCUSSION TITLE
Use api route in next build

### DISCUSSION THREAD

**Discussion by nihiluis** [vercel/next.js#13293] [Help] [answered]
I call an api route in my getStaticProps. How do I make sure the server is started which serves the api route? it's unreachable. I want to use next export later.

```ts
export const getStaticProps = async (ctx: NextPageContext) => {
  const res = await protect<AxiosResponse, AxiosError>(
    axios.get("http://localhost:3334/api/blogEntries")
  )

  let entries: BlogEntry[] = res.ok!.data

  return {
    props: { entries },
  }
}
```

**Comment by nihiluis** [comment]
I just found another issue and it seems it's not supported? How can I then access `fs`? I thought this was the supported way to do it.

**Reply by nihiluis** [reply]
nvm I found the solution here. sorry for the clutter. https://nextjs.org/docs/basic-features/data-fetching

**Comment by jamesmosier** [comment, marked as answer]
The Next docs say:

> You should not fetch an API route from `getStaticProps` — instead, you can write the server-side code directly in `getStaticProps`.

https://nextjs.org/docs/basic-features/data-fetching#write-server-side-code-directly
